ec/src/models/short_weierstrass_jacobian.rs

```diff
@@ -106,7 +106,12 @@ impl<P: Parameters> GroupAffine<P> {
     #[allow(dead_code)]
     pub fn get_point_from_x(x: P::BaseField, greatest: bool) -> Option<Self> {
         // Compute x^3 + ax + b
-        let x3b = P::add_b(&((x.square() * &x) + &P::mul_by_a(&x)));
+        // Rust does not optimise away addition with zero
+        let x3b = if P::COEFF_A == P::BaseField::zero() {
```

Comment:
```suggestion
        let x3b = if P::COEFF_A.is_zero() {
```

Reply:
Is it valid syntax to do something like
```
let mut x3b: P::Basefield;
if (P::COEFF_A.is_zero()) {
    x3b = P::add_b(&(x.square() * &x));
} else {
    x3b = P::add_b(&((x.square() * &x) + &P::mul_by_a(&x)));
}
```
imo assigning to an if is always confusing syntax.